src/server/saved_objects/client/lib/create_id_query.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Finds a document by either its v5 or v6 format
+ *
+ * @param type The documents type
+ * @param id The documents id or legacy id
+**/
+export function createIdQuery({ type, id }) {
+  return {
+    version: true,
+    size: 1,
+    query: {
+      bool: {
+        should: [
+          // v5 document
```

Comment:
@thomasneirynck and I worked through and were able to distil the problem.

A user on 5.6 has the following visualization object in ES:
```
{
  _id: 'foo',
  _type: 'visualization'
}
``` 

They export the saved object, then run the Kibana index migration.

This will result in their ES object being transformed into the following:
```
{
  _id: 'abc123',
  _type: 'doc',
  _source: {
    type: 'visualization',
    legacyId: 'foo'
  }
}
```

If the user then re-imports the export previously made in 5.6 they would be forcing the document to be created with the id specified in order to avoid breaking associations.

```
{
  _id: 'foo',
  _type: 'doc',
  _source: {
    type: 'visualization'
  }
}
```

The current thought is we need to first query for the document before inserting. If a document is returned, and we're forcing the overwrite we will simply update the document. For bulkCreate we would first need to perform a bulkGet on the documents.